docs/config/server-options.md

```diff
@@ -164,17 +162,19 @@ const { createServer: createViteServer } = require('vite')
 async function createServer() {
   const app = express()
 
-  // Create Vite server in middleware mode.
+  // Create Vite server in middleware mode
   const vite = await createViteServer({
-    server: { middlewareMode: 'ssr' }
+    server: { middlewareMode: true },
+    appType: 'custom' // don't include Vite's default HTML handling middlewares
   })
   // Use vite's connect instance as middleware
   app.use(vite.middlewares)
 
   app.use('*', async (req, res) => {
-    // If `middlewareMode` is `'ssr'`, should serve `index.html` here.
-    // If `middlewareMode` is `'html'`, there is no need to serve `index.html`
-    // because Vite will do that.
+    // Since `appType` is `'custom'`, should serve response here.
+    // Note: if `appType` is `'spa'` or `'mpa'`, vite includes middlewares to handle
+    // HTML requests and the `vite404Middleware` so user middlewares should be added
+    // before these to take effect instead
```

Comment:
```suggestion
    // Since `appType` is `'custom'`, should serve response here.
    // Note: if `appType` is `'spa'` or `'mpa'`, Vite includes middlewares to handle
    // HTML requests and 404s so user middlewares should be added
    // before Vite's middlewares to take effect instead
```

Have a few suggestions that I think simplifies/clarifies it a bit.

Reply:
I added the name because a user could use it as a way to inject their middleware in the proper place or remove it. But ya, without context, this "feature" sounds quite strange in the docs. I think we should properly document all the middleware names though and start following semver for them.

Reply:
Sounds like a good idea to me 👍